compiler/jsgen.nim

```diff
@@ -793,6 +793,7 @@ proc lineDir(config: ConfigRef, info: TLineInfo, line: int): Rope =
     rope(toFullPath(config, info)), rope(line), rope(info.toColumn)
   ]
 
+var previousFileName = newStringOfCap(20)  # For frameInfo inside templates.
```

Comment:
make this a field of the `PProc`. And don't use `newStringOfCap`, the results is moved from `toFilename` into the variable anyway. Micro-optimizations require some understanding to apply them properly, you cannot just hack them into a codebase and hope the reviewer misses them. :P